addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1079 +1,1049 @@
-            value: value,
-            boxColor: boxColor,
-        };
-    },
-    /**
-     * This function extracts the information from the data points in tooltipModel.dataPoints
-     * (corresponding to datapoints over a given label determined by the mouse position)
-     * that will be displayed in a custom tooltip.
-     *
-     * @private
-     * @param {Object} tooltipModel see chartjs documentation
-     * @return {Object[]}
-     */
-    _getTooltipItems: function (tooltipModel) {
-        var self = this;
-        var data = this.chart.config.data;
-
-        var orderedItems = tooltipModel.dataPoints.sort(function (dPt1, dPt2) {
-            return dPt2.yLabel - dPt1.yLabel;
-        });
-        return orderedItems.reduce(
-            function (acc, item) {
-                acc.push(self._getTooltipItemContent(item, data));
-                return acc;
-            },
-            []
-        );
-    },
-    /**
-     * Returns the options used to generate chart tooltips.
-     *
-     * @private
-     * @returns {Object}
-     */
-    _getTooltipOptions: function () {
-        var tooltipOptions = {
-            // disable Chart.js tooltips
-            enabled: false,
-            custom: this._customTooltip.bind(this),
-        };
-        if (this.state.mode === 'line') {
-            tooltipOptions.mode = 'index';
-            tooltipOptions.intersect = false;
+
+        _hexToRGBA(hex, opacity) {
+            const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+            const rgb = result.slice(1, 4).map(n => {
+                return parseInt(n, 16);
+            }).join(',');
+            return 'rgba(' + rgb + ',' + opacity + ')';
         }
-        return tooltipOptions;
-    },
-    /**
-     * Returns true iff the current graph can be clicked on to redirect to the
-     * list of records.
-     *
-     * @private
-     * @returns {boolean}
-     */
-    _isRedirectionEnabled: function () {
-        return !this.disableLinking &&
-               (this.state.mode === 'bar' || this.state.mode === 'pie');
-    },
-    /**
-     * Return the first index of the array list where label can be found
-     * or -1.
-     *
-     * @private
-     * @param {Array[]} list
-     * @param {Array} label
-     * @returns {number}
-     */
-    _indexOf: function (list, label) {
-        var index = -1;
-        for (var j = 0; j < list.length; j++) {
-            var otherLabel = list[j];
-            if (label.length === otherLabel.length) {
-                var equal = true;
-                for (var i = 0; i < label.length; i++) {
-                    if (label[i] !== otherLabel[i]) {
-                        equal = false;
+
+        /**
+         * Return the first index of the array list where label can be found
+         * or -1.
+         *
+         * @private
+         * @param {Array[]} list
+         * @param {Array} label
+         * @returns {number}
+         */
+        _indexOf(list, label) {
```

Comment:
It is not used. How is it possible?

Reply:
I don't know the history, but in [master](https://github.com/odoo/odoo/blob/13.0/addons/web/static/src/js/views/graph/graph_renderer.js) it's not used either.
Should I investigate?